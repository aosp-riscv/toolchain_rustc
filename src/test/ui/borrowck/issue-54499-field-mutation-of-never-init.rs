#![warn(unused)]
#[derive(Debug)]
struct S(i32);

type Tuple = (S, i32);
struct Tpair(S, i32);
struct Spair { x: S, y: i32 }

fn main() {
    {
        let t: Tuple;
        t.0 = S(1);
<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
        //~^ ERROR assign to part of possibly uninitialized variable: `t` [E0381]
=======
        //~^ ERROR assign to part of possibly-uninitialized variable: `t` [E0381]
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
        t.1 = 2;
        println!("{:?} {:?}", t.0, t.1);
    }

    {
        let u: Tpair;
        u.0 = S(1);
<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
        //~^ ERROR assign to part of possibly uninitialized variable: `u` [E0381]
=======
        //~^ ERROR assign to part of possibly-uninitialized variable: `u` [E0381]
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
        u.1 = 2;
        println!("{:?} {:?}", u.0, u.1);
    }

    {
        let v: Spair;
        v.x = S(1);
<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
        //~^ ERROR assign to part of possibly uninitialized variable: `v` [E0381]
=======
        //~^ ERROR assign to part of possibly-uninitialized variable: `v` [E0381]
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
        v.y = 2;
        println!("{:?} {:?}", v.x, v.y);
    }
}
