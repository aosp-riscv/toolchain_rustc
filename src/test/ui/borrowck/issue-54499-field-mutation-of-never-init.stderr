<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
error[E0381]: assign to part of possibly uninitialized variable: `t`
  --> $DIR/issue-54499-field-mutation-of-never-init.rs:12:9
   |
LL |         t.0 = S(1);
   |         ^^^^^^^^^^ use of possibly uninitialized `t`

error[E0381]: assign to part of possibly uninitialized variable: `u`
  --> $DIR/issue-54499-field-mutation-of-never-init.rs:20:9
   |
LL |         u.0 = S(1);
   |         ^^^^^^^^^^ use of possibly uninitialized `u`

error[E0381]: assign to part of possibly uninitialized variable: `v`
  --> $DIR/issue-54499-field-mutation-of-never-init.rs:28:9
   |
LL |         v.x = S(1);
   |         ^^^^^^^^^^ use of possibly uninitialized `v`
=======
error[E0381]: assign to part of possibly-uninitialized variable: `t`
  --> $DIR/issue-54499-field-mutation-of-never-init.rs:12:9
   |
LL |         t.0 = S(1);
   |         ^^^^^^^^^^ use of possibly-uninitialized `t`

error[E0381]: assign to part of possibly-uninitialized variable: `u`
  --> $DIR/issue-54499-field-mutation-of-never-init.rs:20:9
   |
LL |         u.0 = S(1);
   |         ^^^^^^^^^^ use of possibly-uninitialized `u`

error[E0381]: assign to part of possibly-uninitialized variable: `v`
  --> $DIR/issue-54499-field-mutation-of-never-init.rs:28:9
   |
LL |         v.x = S(1);
   |         ^^^^^^^^^^ use of possibly-uninitialized `v`
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0381`.
